url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/output/TimeFrameSelector.java

```diff
@@ -0,0 +1,188 @@
+package darwInvest.output;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.geom.Point2D;
+import java.util.ArrayList;
+import java.util.List;
+import darwInvest.environment.Settings;
+import darwInvest.evaluation.BalanceTracker;
+import darwInvest.evaluation.StrategyExecutive;
+import darwin.output.Plot;
+import darwin.output.ScatterPlot;
+import darwin.population.Individual;
+
+public class TimeFrameSelector extends ScatterPlot {
+
+	private static final long serialVersionUID = 6410426510063167109L;
+	private Settings settings;
+	private long activeStart;
+	
+	/**
+	 * Instantiate a new time-frame selector
+	 * @param individual the individual being tested
+	 * @param settings   the settings to use
+	 * @param resolution the number of points to render
+	 * @param selected   the selection listener
+	 */
+	public TimeFrameSelector(Individual individual, Settings settings, int resolution, 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 55 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 43-56 into a separate method.